win32ss/printing/base/winspool/printers.c

```diff
@@ -386,8 +386,38 @@ EndPagePrinter(HANDLE hPrinter)
 BOOL WINAPI
 EnumPrintersA(DWORD Flags, PSTR Name, DWORD Level, PBYTE pPrinterEnum, DWORD cbBuf, PDWORD pcbNeeded, PDWORD pcReturned)
 {
+    WCHAR NameW[255];
+    CHAR  NameA[255];
+    BOOL ret;
+    INT i;
+    PPRINTER_INFO_4W ppi4w = NULL;
+    PPRINTER_INFO_4A ppi4a = NULL;
+
     TRACE("EnumPrintersA(%lu, %s, %lu, %p, %lu, %p, %p)\n", Flags, Name, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
-    return FALSE;
+    if (Name == NULL)
+        NameW[0] = UNICODE_NULL;
+    else
+    /* https://stackoverflow.com/questions/41147180/why-enumprintersa-and-enumprintersw-request-the-same-amount-of-memory */
+        MultiByteToWideChar(CP_ACP, 0, Name, -1, NameW, 255);
+    ret = EnumPrintersW(Flags, NameW, Level, pPrinterEnum, cbBuf, pcbNeeded, pcReturned);
+
+    ppi4w = (PPRINTER_INFO_4W)pPrinterEnum;
+    ppi4a = (PPRINTER_INFO_4A)pPrinterEnum;
+
+    for (i = 0; i < (DWORD)*pcReturned; i++)
```

Comment:
```suggestion
    for (i = 0; i < *pcReturned; i++)
```

Reply:
Dong.

Reply:
pcReturned is a PDWORD, so *pcReturned is a DWORD. If you failed, then there's bug in another place.

Reply:
OK. Now it seems to work fine without the cast. Must have been operator error. Fixed.